### ISSUE TITLE
null error annotating children with errors at stitching

### ISSUE THREAD

**Issue by chess777** [yaacovCR/graphql-tools-fork#26]
I'm using graphql-tools-fork v7.0.3
NodeJS v12.8.1, tried also on v13.0.1
I have a following schema:

```
type Query {
  getType1(): Type1
}

type Type1 {
  type2: [Type2]!
}

type Type2{
  mandatoryField: String!
}
```

If I by error don't fill mandatoryField, then annotateWithChildrenErrors will fail with error:
```
{
  "stack": "TypeError: Cannot set property 'Symbol(subSchemaErrors)' of null\n    at annotateWithChildrenErrors (<path>api\\node_modules\\graphql-tools-fork\\src\\stitching\\errors.ts:73:23)\n    at <path>api\\node_modules\\graphql-tools-fork\\src\\stitching\\errors.ts:68:37\n    at Array.forEach (<anonymous>)\n    at Object.annotateWithChildrenErrors (<path>api\\node_modules\\graphql-tools-fork\\src\\stitching\\errors.ts:68:12)\n    at Object.handleResult (<path>api\\node_modules\\graphql-tools-fork\\src\\stitching\\checkResultAndHandleErrors.ts:63:5)\n    at defaultMergedResolver (<path>api\\node_modules\\graphql-tools-fork\\src\\stitching\\defaultMergedResolver.ts:24:10)\n    at field.resolve (<path>api\\node_modules\\graphql-tools-fork\\src\\stitching\\mergeSchemas.ts:275:16)\n    at field.resolve (<path>api\\node_modules\\graphql-extensions\\src\\index.ts:274:18)\n    at resolveFieldValueOrError (<path>api\\node_modules\\graphql\\execution\\execute.js:467:18)\n    at resolveField (<path>api\\node_modules\\graphql\\execution\\execute.js:434:16)\n    at executeFields (<path>api\\node_modules\\graphql\\execution\\execute.js:275:18)\n    at collectAndExecuteSubfields (<path>api\\node_modules\\graphql\\execution\\execute.js:713:10)\n    at completeObjectValue (<path>api\\node_modules\\graphql\\execution\\execute.js:703:10)\n    at completeValue (<path>api\\node_modules\\graphql\\execution\\execute.js:591:12)\n    at completeValue (<path>api\\node_modules\\graphql\\execution\\execute.js:557:21)\n    at <path>api\\node_modules\\graphql\\execution\\execute.js:492:16",
  "message": "Cannot set property 'Symbol(subSchemaErrors)' of null"
}
```

After validation to annotateWithChildrenErrors comes `object`:
```
{
  type2: [
    null
  ]
}
```
And `childrenErrors`:
```
[
  {
    message: 'Cannot return null for non-nullable field Type2.mandatoryField.',
    path: ['type2', 0, 'mandatoryField']
  }
]
```
As error will be relocated it will reach such state when `object` is `null` and `childrenErrors` will still be an array - this will lead to null exception on line 73.
I've checked the code - v7.1.0 does not have it fixed.

**Comment by yaacovCR** [comment]
Fix only passes along the error if schemas is changed to
```
type Query {
  getType1(): Type1
}

type Type1 {
  type2: [Type2!]! // instead of [Type2]!
}

type Type2{
  mandatoryField: String!
}
```

**Comment by yaacovCR** [comment]
Does that work for now? I believe total fix requires wrapping stitched result instead of annotating it, because you can't annotate null, we get around that by throwing for fields immediately when null, but doesn't work with lists.

**Comment by chess777** [comment]
Thank you! The fix works as you've described.

**Comment by yaacovCR** [comment]
Should be fixed for reals now in 7.1.2.
